src/dsp_tools/xmllib/models/resource.py

```diff
@@ -1173,3 +1182,11 @@ def add_migration_metadata(
             )
         self.migration_metadata = MigrationMetadata(creation_date=creation_date, iri=iri, ark=ark, res_id=self.res_id)
         return self
+
+
+def _check_richtext_before_conversion(value: Any, res_id: str, prop_name: str) -> None:
+    if not is_string_like(value):
+        msg = (
+            f"Resource '{res_id}' has a richtext value that is not a string: " f"Value: {value} | Property: {prop_name}"
```

Comment:
If the value's representation is very long, it hides the information about the property, so I would change the order:

```suggestion
            f"Resource '{res_id}' has a richtext value that is not a string: Property: {prop_name} | Value: {value}"
```